### ISSUE TITLE
How to generate tree view using tree-sitter-tags?

### ISSUE THREAD

**Issue by liuchengxu** [tree-sitter/tree-sitter#1042]
Hi, I'm interesting in intergating tree-sitter in the [vim plugin vista.vim](https://github.com/liuchengxu/vista.vim). Currently, vista.vim already supports LSP and ctags, but the tree view generated by ctags has some issues like https://github.com/liuchengxu/vista.vim/issues/320, I think tree sitter can be a good alternative to ctags. I have tried https://github.com/tree-sitter/tree-sitter/tree/master/tags very quickly, here are my questions:

1. How to convert `syntax_type_id` which is `u32` to a `String` kind?
2. Any guide to generate a tree view for the tree sitter tags?

**Comment by theHamsta** [comment]
Some one made a nvim-treesitter port of vista. But I can't find it anymore. Will give a hint once I found it in our zulip chat. it could also be fairly easily derided from https://github.com/nvim-treesitter/nvim-treesitter/pull/1120. But I suppose you want a solution with an external binary so that it would work in vim also.

**Comment by liuchengxu** [comment]
Yeah, I'm not a fan of neovim only features and want to vim tree-sitter work in vim too, an external binary is on the plan.

**Comment by polyzen** [comment]
> Some one made a nvim-treesitter port of vista. But I can't find it anymore. Will give a hint once I found it in our zulip chat.

[telescope.nvim?](https://github.com/nvim-telescope/telescope.nvim#treesitter-picker)

**Comment by amaanq** [comment]
Closing as stale, as it's not really a tree-sitter specific bug/enhacement/implementation question, and hasn't had an answer in a while. (Discussions might be better, if you still have questions). Also note that the only known consumer of the tags crate is really GitHub.
